redash/authentication/google_oauth.py

```diff
@@ -94,8 +107,7 @@ def authorized():
 
     if not verify_profile(org, profile):
         logger.warning("User tried to login with unauthorized domain name: %s (org: %s)", profile['email'], org)
-        flash("Your Google Apps account ({}) isn't allowed.".format(profile['email']))
-        return redirect(url_for('redash.login', org_slug=org.slug))
+        return redirect(url_for('redash.login', relogin='1', org_slug=org.slug))
```

Comment:
This can't redirect back to `redash.login`, the `redash/handlers/authentication.py::login` function, but back to either the [`org_login`](https://github.com/getredash/redash/blob/93b34bcbce2b3df328e1c1fac4261fa83e8717b3/redash/authentication/google_oauth.py#L60-L63) or [`login`](https://github.com/getredash/redash/blob/93b34bcbce2b3df328e1c1fac4261fa83e8717b3/redash/authentication/google_oauth.py#L66-L72) views in the `redash/authentication/google_auth.py` file. Otherwise the `relogin` parameter isn't picked up since it's lost when `redash.login` renders the button with the link to the Google login again.

I'm wondering if there is a chance that this ends in an endless loop, but I think Google will halt that when it happens too often.

The `org_login` view will need to also pass the relogin paramter along to `login` since it's really just a small facade to populate the `org_slug` URL path parameter.

Alternatively, we can also modify the `redash.login` main view (/login) and check if relogin was provided to it and pass it along the the Google auth link when it renders the button. That would mean that if users need to relogin, they'd always end up on the main login page with the button again, instead of being passed to Google right away.

Either way, this needs another pass, I'm afraid.